Samples~/Editor/SoftDeleteEdgesAction.cs

```diff
@@ -0,0 +1,98 @@
+// This script demonstrates how to create a new action that can be accessed from the ProBuilder toolbar.
+// A new menu item is registered under "Geometry" actions called "Remove edges soft".
+
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.ProBuilder.MeshOperations;
+
+namespace UnityEditor.ProBuilder.Actions
+{
+	/// <summary>
+	/// This is the actual action that will be executed.
+	/// </summary>
+	[ProBuilderMenuAction]
+	public class SoftDeleteEdgesAction : MenuAction
+	{
+		public override ToolbarGroup group { get { return ToolbarGroup.Geometry; } }
+		public override Texture2D icon { get { return null; } }
+		public override TooltipContent tooltip { get { return k_Tooltip; } }
+
+		/// <summary>
+		/// What to show in the hover tooltip window.
+		/// TooltipContent is similar to GUIContent, with the exception that it also includes an optional params[]
+		/// char list in the constructor to define shortcut keys (ex, CMD_CONTROL, K).
+		/// </summary>
+		static readonly TooltipContent k_Tooltip = new TooltipContent
+		(
+			"Remove edges soft",
+			"Remove selected edges and merge faces."
+		);
+
+		/// <summary>
+		/// Determines if the action should be enabled or grayed out. Here when at least one edge is selected.
+		/// </summary>
+		/// <returns></returns>
+		public override bool enabled
+		{
+			get { return MeshSelection.selectedEdgeCount > 0; }
+		}
+
+		/// <summary>
+		/// This action is applicable in Edge selection modes.
+		/// </summary>
+		public override SelectMode validSelectModes
+		{
+			get { return SelectMode.Edge; }
+		}
+
+		/// <summary>
+		/// Return a pb_ActionResult indicating the success/failure of action.
+		/// </summary>
+		/// <returns></returns>
+		public override ActionResult DoAction()
+		{
+			var selection = MeshSelection.top.ToArray();
+			Undo.RecordObjects(selection, "Removing edges");
+			
+			List<SimpleTuple<Vector3, Vector3>> edgesBounds;
+			foreach (ProBuilderMesh pbMesh in selection)
+			{
+				if(pbMesh.selectedEdgeCount > 0)
+				{
+					var selectedEdges = pbMesh.selectedEdges;
+					
+					edgesBounds = selectedEdges.Select(edge => new SimpleTuple<Vector3,Vector3>(pbMesh.positions[edge.a], pbMesh.positions[edge.b])).ToList();
+					
+					for(int i= 0; i < edgesBounds.Count; i++)
+					{	
+						List<Face> facesToMerge = pbMesh.faces.Where(face => face.edges.ToList().Exists(e => IsEdgeEquivalent(pbMesh, e, edgesBounds[i]))).ToList();
+						Face f = MergeElements.Merge(pbMesh, facesToMerge);
```

Comment:
We could make `ElementSelection.GetNeighborFaces` public and use that function instead. It would be a good idea to add a non-allocating overload of that function as well.

Reply:
It would be nice if we can dodge having to ToList the ReadOnlyArray by having an internal function that can use edgesInternal instead. Also, having a public and optimized function that gets face neighbors with a NonAlloc result overload seems like a good idea. 

In context, I think the line is perfectly fine, but considering that this is a common enough thing to do that people would probably copy/paste this from the sample and not understand why their stuff, in a more critical path, might be slow.  I don't think we should be optimizing the samples but any stuff that requires linq to be clean could point to a lack in our public API.

Reply:
@karl- and @antoine-unity 
I did try to make this as you were saying, feel free to let me know your thoughts on this.
The code is way more dense than before as this 'simple' case a little more complicated than expected at first sight, OR I might be missing some info and in that case, please give me an update on this :) :
- we cannot merge every single couple of face designated by an edge directly as it'll mess up the rest of the selection => _selected edges will be modified by the merge_
- we don't want to merge all faces designated by the edges together => _you want to remove edges between faces, not merge all faces together_
- we do want to create a group of face to merge together if edge selection contains enough edges to connect them

This new method take all of this into account.
The previous one was doing the same too by reasoning on Vertices positions (Vector3) instead of indexes

Let me know your thoughts on that and if you have a simpler solution to handle this :)